test/tape/connection-config-provider.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
```

Comment:
Can you also add integration test that would be run on all databases and verify connection settings to be reused across multiple connections?

Reply:
The current test does this for psqlite3 - creating 2 concurrent connections and counting the number of provider invocations. Just 1 invocation means the provider has been reused.

Is there a generic way to write a test to run on top of all databases?

Reply:
Yeah, check tests within `https://github.com/knex/knex/tree/master/test/integration` folder. We basically iterate over each test and pass `knex` instance to it with initialized connection. I assume you can reuse the code from how this iteration is done, but probably you will need to instantiate it separately, as you need different configuration.

Reply:
@oranoran You can see how iteration is done here: https://github.com/knex/knex/blob/master/test/integration/index.js